Master adis1649x

stefpopa opened the pull request:
The main purpose of this pull request is to add support for adis1649x family of devices in the already existing adis16480 driver, but on the way towards this goal, it also deals with some outstanding items:
* Makes the drdy pin configurable
* Add OF device ID table
* Add missing docs

mhennerich commented:
One comment - once we send this upstream. We should also send this patch, too.

https://github.com/analogdevicesinc/linux/commit/b935e10c3f9f61c1abb59dda983b8618d4656e74

mirceacaprioru commented on drivers/iio/imu/adis16480.c:
```diff
@@ -837,6 +856,58 @@ static const struct adis_data adis16480_data = {
 	.enable_irq = adis16480_enable_irq,
 };
 
+
```
nitpick: extra line

mirceacaprioru commented on drivers/iio/imu/adis16480.c:
```diff
@@ -837,6 +856,58 @@ static const struct adis_data adis16480_data = {
 
+
+static int adis16480_config_irq_pin(struct device_node *of_node,
+		struct adis16480 *st)
```
this indentation looks inconsistent.

stefpopa commented:
@mhennerich: ack

mirceacaprioru commented on drivers/iio/imu/adis16480.c:
```diff
@@ -984,9 +984,19 @@ static const struct spi_device_id adis16480_ids[] = {
 };
 MODULE_DEVICE_TABLE(spi, adis16480_ids);
 
+static const struct of_device_id adis16480_of_match[] = {
```
nitpick There is a misspelling in the commit message devicdevices

commodo commented on drivers/iio/imu/adis16480.c:
```diff
@@ -837,6 +856,58 @@ static const struct adis_data adis16480_data = {
+	pin = ADIS16480_PIN_DIO2;
+	for (i = 0; i < ARRAY_SIZE(adis16480_int_pin_names); i++) {
+		irq = of_irq_get_byname(of_node, adis16480_int_pin_names[i]);
+		if (irq > 0)
```
do we need to `break` early here in certain conditions ?
it looks like the `pin` variable could be re-assigned a few times, and only the last one is used

stefpopa replied:
The point of using interrupt-names is to determine which DIO pin should be used as an interrupt. So, it should be something like: `interrupt-names = "DIO2";`. 
However, I think it is possible to assign multiple values to it: `interrupt-names = "DIO2", "DIO1", "DIO0";`. In this example,  I think that DIO0 will be selected. 
But, the question is, which one does the user actually want to be selected?

mirceacaprioru commented:
Nice job overall.

commodo commented on drivers/iio/imu/adis16480.c:
```diff
@@ -124,6 +124,7 @@ struct adis16480_chip_info {
 	unsigned int gyro_max_scale;
 	unsigned int accel_max_val;
 	unsigned int accel_max_scale;
+	unsigned int temp_max_scale;
```
this patch is fine as-is;
but i do suspect upstream will cringe a bit and say something like "i wish these were 2 patches" :)

for our case, i'm fine with having this as-is

stefpopa replied:
:), you are probably right

stefpopa commented:
changes in v2:
* added missing brackets 
* fixed indentation inconsistencies
* added a `break `statement when the irq name is found
* fixed spelling in the commit message
* added a separate commit which deals with temperature max scale in a generic way before adding support for new devices.

commodo commented:
[![LGTM](https://i.redd.it/j1gei60ye9h21.jpg)](https://i.redd.it/j1gei60ye9h21.jpg)
